ex4B/ex4.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+
+const CODE_FOR_0 = "0".charCodeAt();
+
+function toNumber(str) {
+	var strNum = '', strDec = '', number = null, i = 0, seenDot = false, negative = false;
```

Comment:
Get into the habit of only declaring the `var`s when needed -- this adds to clarify (because you know what section of the code pertains to what variable, and also it helps seeing what the initial value is in the right context).